lib/shared/bucketing-assembly-script/assembly/types/target.ts

```diff
@@ -172,6 +171,24 @@ export class AudienceFilterOrOperator extends JSON.Value {
     }
 }
 
+export class CustomDataFilter extends AudienceFilterOrOperator {
+    dataKeyType: string
+    dataKey: string
+
+    constructor(filter: JSON.Obj) {
+        super(filter)
+        this.dataKey = getStringFromJSON(filter, 'dataKey')
+        this.dataKeyType = isValidString(filter, 'dataKeyType', validDataKeyTypes)
```

Comment:
how is this different than the constructor above? 
<img width="797" alt="image" src="https://user-images.githubusercontent.com/1219069/163210422-f6a04bce-af7d-4a84-9cf6-a15369af44cc.png">

Reply:
ahh so it makes it non-optional?

Reply:
yeah, its basically just for input data validation because it still gets passed around as an AudienceOrOperator object. I'm then explicitly casting it to the CustomDataFilter when it gets used as that, which isn't great. Can't find a better way to do it though due to the lack of union types